bundles/org.openhab.binding.somfytahoma/src/main/java/org/openhab/binding/somfytahoma/internal/handler/SomfyTahomaBridgeHandler.java

```diff
@@ -364,9 +372,8 @@ private String getFormattedParameters(Collection<String> stateNames) {
 
     private void getTahomaUpdates() {
         logger.debug("Getting Tahoma Updates...");
-        if (ThingStatus.OFFLINE.equals(thing.getStatus())) {
-            logger.debug("Doing relogin");
-            login();
+        if (ThingStatus.OFFLINE.equals(thing.getStatus()) && !reLogin()) {
```

Comment:
Because `ThingStatus` is an enum so you can use `==` in comparisons:

```suggestion
        if (ThingStatus.OFFLINE == thing.getStatus() && !reLogin()) {
```

Reply:
changed